src/coreclrDebug/activate.ts

```diff
@@ -290,9 +301,17 @@ export class DebugAdapterExecutableFactory implements vscode.DebugAdapterDescrip
                 'vsdbg-ui' + CoreClrDebugUtil.getPlatformExeExtension()
             );
 
-            // Look to see if DOTNET_ROOT is set, then use dotnet cli path
-            const dotnetRoot: string =
-                process.env.DOTNET_ROOT ?? (dotNetInfo.CliPath ? path.dirname(dotNetInfo.CliPath) : '');
+            // Look to see if DOTNET_ROOT is set, then use dotnet cli path for omnisharp, then hostExecutableInfo
+            let dotnetRoot: string | undefined = process.env.DOTNET_ROOT;
+            if (!dotnetRoot) {
+                if (dotNetInfo.CliPath) {
+                    dotnetRoot = path.dirname(dotNetInfo.CliPath);
+                } else if (hostExecutableInfo.path) {
```

Comment:
>if (hostExecutableInfo.path) { [](http://example.com/codeflow?start=23&length=30)

I think you need some more checks here:
1. If the architecture of the debugger != the architecture of this runtime, we don't want to use it
2. If there already is a global .NET runtime, we don't want to use it

Reply:
> If there already is a global .NET runtime, we don't want to use it

Whats the best way to check for this? I thought this was covered with process.env.DOTNET_ROOT

Reply:
> We might need to re-write getHostExecutableInfo to get the correct architecture.
I think you just want to ignore it/not compute it if we are trying to get a different architecture. In other words, if you want to target `x64`, but you are on an ARM64 mac, you need to download the the .NET Runtime yourself

Reply:
> Whats the best way to check for this? I thought this was covered with process.env.DOTNET_ROOT

Normally we wouldn't use `DOTNET_ROOT` if there was a globally installed .NET SDK. I would think you would want to exec `dotnet --version` and if that succeeded: we use the global version.